apps/web/app/helpers/array-data.ts

```
@@ -98,8 +97,11 @@ const formatTime = (d: Date | string, addHour: boolean) => {
  * @returns An object where the keys are the unique values of the key function and the values are arrays of items.
  */
 export const groupBy = <T, K extends keyof any>(array: T[], key: (item: T) => K): Record<K, T[]> =>
-	array.reduce((acc, item) => {
-		const groupKey = key(item);
-		(acc[groupKey] ||= []).push(item);
-		return acc;
-	}, {} as Record<K, T[]>);
+	array.reduce(
+		(acc, item) => {
+			const groupKey = key(item);
+			(acc[groupKey] ||= []).push(item);
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)